notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+    "train_data.info()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data[\"family\"] = train_data[\"family\"].str.lower()     # for easier reading \n",
+    "train_data.describe(percentiles=[.25, 0.375, .5, .75, .875]).drop([\"count\"])       \n",
+    "\n",
+    "# I dropped \"count\" row to get rid of exponentioal number presentation\n",
+    "# And added .875 percentile to understand approximately how many objects have no onpromotion"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "6yDCEDYYGveo",
+    "outputId": "839affcc-2928-4892-c798-459ff56b59f0"
+   },
+   "outputs": [],
+   "source": [
+    "train_data[\"family\"].describe()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>The only categorical feature (in this table) \"family\" has 33 possible unique values (not so a lot) <br>\n",
+    "That means we can easily use one-hot-encoding during model training</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "family_values = train_data[\"family\"].unique()\n",
+    "family_values"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 40))\n",
+    "for i, value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(len(family_values)//3, 3, i+1)\n",
+    "  current_value_data = train_data[train_data[\"family\"] == value].groupby([\"date\"]).mean()\n",
+    "  plt.xlim=(0, 1750)\n",
+    "  plt.ylim=(0, current_value_data[\"sales\"].max())\n",
+    "  plt.scatter(x=np.arange(len(current_value_data.index)), y=current_value_data[\"sales\"])\n",
+    "  plt.title(value)\n",
+    "plt.xlim=(0, 1750)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>Actually, these plots don't really help us, because, looking on them, we can only catch some common \"trends\", which describe global changes in some products sales.<br>For example, we can be sure that on the 1th January every year number of sales is equal to 0 (actually, it is necessary to check, but I think it is obvious). <br> Also we can say that the number of sales in general has a positive dynamic (for example, 'automotive', 'bread/bakery', 'grocery i', 'personal care', ...).<br>Some of the product types have a negative dynamic (such as lingerie).<br>Some goods have very interesting sales distribution (books, produce, froxen foods, ladieswear, ...), and we can't say right now, what is the reason of that.<br>Also some goods families have seasonal increase in sales('school and office supplies', 'liquor, wine, beer', 'grocery ii', 'frozen foods'). </i>"
```

Comment:
Indeed, these plots are not so informative as they might be. To investigate trends it's better to plot probably rolling median (as it is more robust to outliers than the mean) over short time period: week or month. This can be achieved with the code like `df.groupby(pd.Grouper('date', freq='W'))['sales'].median()` (or `freq=M` for monthly aggregations). And using ordinary `plot` instead of `scatterplot`.

> Some of the product types have a negative dynamic (such as lingerie).

> Some goods have very interesting sales distribution (books, produce, froxen foods, ladieswear, ...), and we can't say right now, what is the reason of that.

These would be good points to discuss with your potential customer (if it was a real project).

On these pictures I would additionally point your attention to outliers, like in the example below:
![image](https://user-images.githubusercontent.com/24436463/179364504-9170646b-faa1-46e1-b222-93b28a946dfd.png)
What dates / events correpond to them? Probably some holidays (maybe even the New Year you mentioned)? Should we do something with such samples when applying an ML model?

As for books category, I suppose it was new at some point which led to short-time extraordinary demand. But still it's interesting to know which date corresponds to the begginning of sales.

Also there is an interesting drop in the middle of sales that repeats for multiple categories:
![image](https://user-images.githubusercontent.com/24436463/179364861-2bcddbbd-03be-4171-8db5-5b03c0c0a15f.png)
What happened here?